src/github.com/couchbase/sync_gateway/rest/routing.go

```diff
@@ -251,6 +265,22 @@ func wrapRouter(sc *ServerContext, privs handlerPrivs, router *mux.Router) http.
 	})
 }
 
+func matchedOrigin(allowOrigins []string, rqOrigins []string) string {
+	for _, av := range allowOrigins {
+		for _, rv := range rqOrigins {
```

Comment:
Would there be a performance benefit to flipping these - iterate over rqOrigins in the outer loop, since that's the set we're trying to match?